### ISSUE TITLE
False positive and unsafe correction with `Style/HashAsLastArrayItem` and double splat operator.

### ISSUE THREAD

**Issue by inkstak** [rubocop/rubocop#11161]
## Expected behavior

Using the double splat operator is very convenient to avoid adding an empty hash to an array.

```ruby
def build_args(options = {})
  [1, 2, 3, **options]
end

> build_args
 => [1, 2, 3]

> build_args(foo: :bar)
 => [1, 2, 3, {:foo=>:bar}]
```

## Actual behavior

With the default rubocop settings:

```yaml
Style/HashAsLastArrayItem:
  EnforcedStyle: braces
```

The actual behavior reports an offense.

```
example.rb:2:13: C: [Correctable] Style/HashAsLastArrayItem: Wrap hash in { and }. (https://rubystyle.guide#hash-literal-as-last-array-item)
  [1, 2, 3, **options]
            ^^^^^^^^^
```

It also autocorrects the statement to change the behavior:

```ruby
def build_args(options = {})
  [1, 2, 3, { **options }]
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 2.7.6) +server [arm64-darwin21]
  - rubocop-performance 1.15.0
  - rubocop-rails 2.17.2
  - rubocop-rspec 2.14.2
```

**Comment by inkstak** [comment]
Thanks !
